src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotDeclareVisibleInstanceFields.cs

```diff
@@ -42,11 +42,11 @@ public override void Initialize(AnalysisContext analysisContext)
             {
                 var field = (IFieldSymbol)obj.Symbol;
 
-                // Only report diagnostic on non-static, non-const, public fields
+                // Only report diagnostic on non-static, non-const, public or protected or protected internal/friend fields.
                 // Additionally, only report on members within externally visible types for FxCop compat.
                 if (!field.IsStatic &&
```

Comment:
This whole condition should be simplified to: `if (!field.IsStatic && !field.IsConst && field.IsExternallyVisible())`

Reply:
Done.